python/torch_mlir/__init__.py

```diff
@@ -248,6 +248,64 @@ def _get_for_tracing(
 }
 
 
+def canon_extra_library(extra_library):
+    extra_library_file_name = ""
+    if len(extra_library) != 0:
+        extra_library_dict = {}
+        for library_func in extra_library:
+            extra_library_dict[library_func.__name__] = library_func
+        mlir_library = generate_library(extra_library_dict)
+
+        extra_library_file_name = \
+            tempfile.gettempdir() + "/custom_op_extra_library.mlir"
+        with open(extra_library_file_name, "w") as f:
+            f.write(mlir_library)
+    return extra_library_file_name
+
+
+def lower_mlir_module(verbose, output_type, module):
```

Comment:
Why does this need to be exposed?

Reply:
There's often a lot of talk of "exposure" in these review and I have to finally concede I have no idea what it means to expose/not-expose things. I'm sure you're well aware that you can't make private absolutely anything in Python. So what's the material recommendation here? That I should prefix these functions with the `_` fig leaf?

Reply:
The "exposure" is about managing user expectations -- without clear guidance (or mechanical enforcement, which as you pointed out is hard in Python) users can end up relying on things that we don't intend to support, which can cause grief (either on their side, via us breaking them, or our side, by us getting stuck supporting something that we didn't intend to). It is also a useful signal to us for users to come and say that they want particular API's to be exposed so that they can officially rely on them.

Indeed, using `__all__` would be great here. So far we have been using the leading `_` convention.